pkg/api/config.go

```diff
@@ -3,71 +3,177 @@ package api
+		}
+		if testConfig := test.OpenshiftInstallerClusterTestConfiguration; testConfig != nil {
+			typeCount++
+			validationErrors = append(validationErrors, validateTargetCloud(fmt.Sprintf("%s[%d]", fieldRoot, num), testConfig.TargetCloud))
+		}
+		if testConfig := test.OpenshiftInstallerSmokeClusterTestConfiguration; testConfig != nil {
+			typeCount++
+			validationErrors = append(validationErrors, validateTargetCloud(fmt.Sprintf("%s[%d]", fieldRoot, num), testConfig.TargetCloud))
+		}
+		if typeCount == 0 {
+			// TODO remove when the migration is completed
+			if len(test.From) == 0 {
+				validationErrors = append(validationErrors, fmt.Errorf("%s[%d] has no type", fieldRoot, num))
 			}
-			if typeCount == 0 {
-				// TODO remove when the migration is completed
-				if len(test.From) == 0 {
-					errs = append(errs, fmt.Errorf("test %q has no type", test.As))
-				}
-			} else if typeCount == 1 {
-				// TODO remove when the migration is completed
-				if len(test.From) > 0 {
-					errs = append(errs, fmt.Errorf("test %q specifies both `From` and a test type", test.As))
-				}
-			} else if typeCount > 1 {
-				errs = append(errs, fmt.Errorf("test %q has more than one type", test.As))
+		} else if typeCount == 1 {
+			// TODO remove when the migration is completed
+			if len(test.From) > 0 {
+				validationErrors = append(validationErrors, fmt.Errorf("%s[%d] specifies both `From` and a test type", fieldRoot, num))
 			}
+		} else if typeCount > 1 {
+			validationErrors = append(validationErrors, fmt.Errorf("%s[%d] has more than one type", fieldRoot, num))
 		}
 	}
-	buildRootImage := config.InputConfiguration.BuildRootImage
-	if buildRootImage != nil {
-		if buildRootImage.ProjectImageBuild != nil && buildRootImage.ImageStreamTagReference != nil {
-			errs = append(errs, errors.New("both git_source_image and image_stream_tag cannot be set for the build_root"))
-		} else if buildRootImage.ProjectImageBuild == nil && buildRootImage.ImageStreamTagReference == nil {
-			errs = append(errs, errors.New("you have to specify either git_source_image or image_stream_tag for the build_root"))
-		}
+	return kerrors.NewAggregate(validationErrors)
+}
+
+func validateImageStreamTagReference(fieldRoot string, input ImageStreamTagReference) error {
+	var validationErrors []error
+
+	if _, err := url.ParseRequestURI(input.Cluster); err != nil && input.Cluster != "" {
+		validationErrors = append(validationErrors, fmt.Errorf("%s.cluster invalid URL given: %s", fieldRoot, err))
+	}
+
+	if input.Tag == "" {
+		validationErrors = append(validationErrors, fmt.Errorf("%s.tag: value required but not provided", fieldRoot))
+	}
+
+	return kerrors.NewAggregate(validationErrors)
+}
+
+func validateImageStreamTagReferenceMap(fieldRoot string, input map[string]ImageStreamTagReference) error {
+	var validationErrors []error
+	// map has to be sorted to avoid flakes in tests.
```

Comment:
If we're not trying to validate a specific error message, is this still true?

Reply:
No, it can be removed. The errorMessage have been removed [here](https://github.com/openshift/ci-operator/commit/ef19d3ab9d5cad8f2199873b85d112a44e2b2aa7#diff-787bc42cef1afcb577450319f808c3b2L12)